### PR TITLE
Hardcode Rework

### PR DESCRIPTION
**Pull request by blackcrystall**
# About the pull request

Added ECHO support to sounds (byond function implementation to be accessible), some changes from one letter vars.

Rework of admin shoot the ship pannel, so now it's not hardoced

# Explain why it's good for the game

Just backend rework, almost same

:cl: BlackCrystalic
add: Byond backed function of sound ECHO ported
refactor: refactor of weaponhits hardcode
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/almayer/weaponhits.dm [hunk 1]
```diff
@@ -1,123 +1,145 @@
...
+	var/list/echo_list = new(18)
+	echo_list[ECHO_OBSTRUCTION] = -2500
+	if(intercepted)
+		playsound(picked_atom, pick(intercept_sound), 100, 1, 100, echo = echo_list)
+	else
+		playsound(picked_atom, pick(miss_sound), 5, 1, 100, echo = echo_list)
+	shipwide_ai_announcement("[capitalize(name)] [intercepted ? "INTERCEPTED" : "MISSED"]!", MAIN_AI_SYSTEM, 'sound/effects/double_klaxon.ogg')
+
+/datum/space_weapon_ammo/rail_gun/hit_target(picked_atom)
+	var/list/echo_list = new(18)
```
Comment (harryob) [inline code/modules/almayer/weaponhits.dm:85]:
```suggestion
	var/list/echo_list = new /list(18)
```

**Review by harryob — CHANGES_REQUESTED** [review]
otherwise looks fine

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by harryob — APPROVED** [review]

**Pull request merged**